common/persistence/cassandra/metadata_store.go

```diff
@@ -161,22 +168,35 @@ func (m *MetadataStore) CreateNamespaceInV2Table(
 
 	previous := make(map[string]interface{})
 	applied, iter, err := m.session.MapExecuteBatchCAS(batch, previous)
+	deleteOrphanNamespace := func() {
+		// Delete namespace from `namespaces_by_id`
+		if errDelete := m.session.Query(templateDeleteNamespaceQuery, request.ID).WithContext(ctx).Exec(); errDelete != nil {
+			m.logger.Warn("Unable to delete orphan namespace record. Error", tag.Error(errDelete))
+		}
+	}
+
 	if err != nil {
 		return nil, serviceerror.NewUnavailable(fmt.Sprintf("CreateNamespace operation failed. Inserting into namespaces table. Error: %v", err))
 	}
+
 	defer func() { _ = iter.Close() }()
 
 	if !applied {
-		// Namespace already exist.  Delete orphan namespace record before returning back to user
-		if errDelete := m.session.Query(templateDeleteNamespaceQuery, request.ID).WithContext(ctx).Exec(); errDelete != nil {
-			m.logger.Warn("Unable to delete orphan namespace record. Error", tag.Error(errDelete))
-		}
+		if id, ok := previous["id"]; ok {
```

Comment:
if the batch fails due to notification_version in namespace metadata row can't be updated, the `ok` variable here will be `false` (I mean instead of `true` and an empty `id`)?

Reply:
hmmm, is there a way to detect that case? maybe through `iter` returned by the batch?

Reply:
you can check if the `name` value for the conflicting row is `namespaceMetadataRecordName`